types/index.d.ts

```diff
@@ -13,182 +13,75 @@
-     *
-     * This is to make sure transforms and formats always have the unmodified version of the original property.
-     */
-    original: Property;
-    /**
-     * The file path of the file the token is defined in.
-     *
-     * This file path is derived from the source or include file path arrays defined in the configuration.
-     */
-    filePath: string;
-    /**
-     * If the token is from a file defined in the source array as opposed to include in the [configuration](https://amzn.github.io/style-dictionary/#/config).
-     */
-    isSource: boolean;
-    attributes: Attributes;
-    [key: string]: any;
-  }
-
-  interface NameTransform {
-    type: "name";
-    matcher?: (prop: Prop) => boolean;
-    transformer: (prop: Prop, options?: Platform) => string;
-  }
-
-  interface ValueTransform {
-    type: "value";
-    transitive?: boolean;
-    matcher?: (prop: Prop) => boolean;
-    transformer: (prop: Prop, options?: Platform) => string;
-  }
-
-  interface AttributeTransform {
-    type: "attribute";
-    matcher?: (prop: Prop) => boolean;
-    transformer: (prop: Prop, options?: Platform) => { [key: string]: any };
-  }
-
-  type Transform = NameTransform | ValueTransform | AttributeTransform;
-
-  interface Transforms {
-    [name: string]: Transform;
-  }
-
-  interface TransformGroup {
-    name: string;
-    transforms: string[];
-  }
-
-  interface TransformGroups {
-    [name: string]: string[];
-  }
-
-  type Formatter = (this: File, dictionary: Core, config: Platform) => string;
-
-  interface Format {
-    name: string;
-    formatter: Formatter;
-  }
-
-  interface Formats {
-    [name: string]: Formatter;
-  }
-
-  interface Action {
-    do(dictionary: Core, config: Platform): void;
-    undo?(dictionary: Core, config: Platform): void;
-  }
+import _DesignToken, {DesignTokens as _DesignTokens} from './DesignToken';
+import _File from './File';
+import _Options from './Options';
+import _FileHeader from './FileHeader';
+import _TransformedToken, {TransformedTokens as _TransformedTokens} from './TransformedToken';
+import _Platform from './Platform';
+import _Transform from './Transform';
+import _Filter from './Filter';
+import _Format from './Format';
+import _Dictionary from './Dictionary';
+import _TransformGroup from './TransformGroup';
+import _Action from './Action';
+import _Parser from './Parser';
+import _Config from './Config';
+import FormatHelpers from './FormatHelpers';
+
+import { Named, Keyed } from './_helpers';
 
-  interface Actions {
-    [name: string]: Action;
-  }
-
-  type Matcher = (prop: Prop) => boolean;
-
-  interface Filter {
-    name: string;
-    matcher: Matcher;
-  }
-
-  interface Filters {
-    [name: string]: Matcher;
-  }
-
-  interface ParserOptions {
-    contents: string;
-    filePath: string;
-  }
-
-  interface Parser {
-    pattern: RegExp;
-    parse: (props: ParserOptions) => Properties;
-  }
+declare namespace StyleDictionary {
```

Comment:
Just a note, the TypeScript docs recommend "modules over namespaces in modern code."

https://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html#using-modules

To transform this file, you would need to 

delete the namespace wrapper

```diff
- declare namespace StyleDictionary {
- }
```

update the types inside it to be exported

```diff
- type DesignToken = _DesignToken;
+ export type DesignToken = _DesignToken; 
```

Or, if you also removed the default exporting of types in many of the files, you could replace them so you just reexport the original types.

```diff
- import _File from "./File";
- type File = _File;
+ export { File } from "./File";
```

and finally update the main export.

```diff
- declare var StyleDictionary: StyleDictionary.Core;
- export = StyleDictionary;
+ declare var StyleDictionary: Core;
+ export default StyleDictionary;
```

Reply:
I knew doing this felt so wrong:

```typescript
import _File from "./File";
type File = _File;
```

Reply:
I think we might need to revert back to the namespace style or else it might mess up the autocomplete/intellisense for JS users.